src/backend/api/meals.js

```diff
@@ -2,14 +2,213 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+// /api/meals	GET	Returns all meals
+
 router.get("/", async (request, response) => {
+  let query = knex("meal");
+
+  if (
+    "maxPrice" in request.query &&
+    typeof request.query.maxPrice === "string"
+  ) {
+    query = query.whereRaw("price <= ?", [request.query.maxPrice]);
+  }
+
+  if (request.query.availableReservations) {
+    query = query.join("reservation", "meal.id", "=", "reservation.meal_id");
```

Comment:
This just joins the 2 tables but you should be adding some check that checks if there are available reservations or not, depending on if `?availableReservations=true` or `?availableReservations= false`.

Reply:
hm, can you elaborate? My thinking was that if there is a reservation.meal_id - then there is a reservation. I added this `if ((request.query.availableReservations = true)) {
    query = query.join("reservation", "meal.id", "=", "reservation.meal_id");
  }`
  If `availableReservations= false` should I send 404?

Reply:
Just joining is not enough because you should compare the number of guests (sum of guests for all meal reservations) who have reserved against the max number of guests (available on meals record).

Then
- `availableReservations=true`, it means there are room for more guests.
- `availableReservations=false`, it means the meal is fully booked up.